src/backend/utils/url-parser.js

```diff
@@ -0,0 +1,17 @@
+// parsesUrl to make sure there is only a single port
+const parseUrl = (url, port) => {
+  let result;
+  try {
+    const urlObj = new URL(url);
+    urlObj.port = port;
+    if (urlObj.port !== '') {
```

Comment:
Why bother with this?  Consider:

```js
parseUrl('http://localhost'); // no port (undefined or null), `port = undefined` or `port = null` will both give ''
parseUrl('http://localhost', 3000); // port as number, `port = 3000` will give '3000'
parseUrl('http://localhost', '3000'); // port as string, `port = '3000'` will give '3000'
parseUrl('http://localhost', 'invalid'); // port is invalid, `port = 'invalid'` will give ''
```

Given that `new URL(...).port = ...` won't pollute the port (it will just ignore bad data), shouldn't we always set `urlObj.port = port` and then return `urlObj.href` immediately?

Reply:
is this about the try block ?

Reply:
@humphd  I don't understand this comment, is this not what I am already doing

Reply:
No, you only return the `.href` in the case that the port is not nothing, meaning you now have 3 different return types:

1. `string` when port is set and we get back `.href`
2. `undefined` when port is `''`
3. `null` when there's a parsing error.

Let's reduce that to 2 and return either `null` or the URL string.  I don't think you need this `if` check, just return `urlObj.ref` in all cases within this try.